pkgs/development/interpreters/php/default.nix

```diff
@@ -227,7 +246,10 @@ let
       '';
 
       src = fetchurl {
-        url = "https://www.php.net/distributions/php-${version}.tar.bz2";
+        urls = [
+          "https://www.php.net/distributions/php-${version}.tar.bz2"
+          "https://downloads.php.net/~derick/php-${version}.tar.bz2"
```

Comment:
This was a comment that came up earlier I noticed as well. But I think we should look into moving to their github mirror: https://github.com/php/php-src/releases

They publish all releases there, both stable and unstable. So we could (probably) use fetchFromGitHub instead and pass the version as revision and the sha256 as before.

Reply:
With fetchFromGitHub error installPhase
```
Installing PEAR environment:      /nix/store/6m1p2hqbff6h863gxgclms3jfydx47hz-php-7.2.24/lib/php/

Warning: fopen(): php_network_getaddresses: getaddrinfo failed: Name or service not known in /build/source/pear/fetch.php on line 66

Warning: fopen(https://pear.php.net/install-pear-nozlib.phar): failed to open stream: php_network_getaddresses: getaddrinfo failed: Name or service not known in /build/source/pear/fetch.php on line 66

Error..
fopen(https://pear.php.net/install-pear-nozlib.phar): failed to open stream: php_network_getaddresses: getaddrinfo failed: Name or service not known
make: *** [Makefile:512: install-pear] Error 1
```

Reply:
It happens that the release tarballs are different from the tags. And fetchFromGitHub fetches the tags.

You could try to fetchurl `https://github.com/php/php-src/archive/php-${version}.tar.gz` instead. It may yield a different result and you probably need to change the hash to get this different result.

Reply:
Well then... Looks like we can't build releases from github that easily :disappointed:

Otherwise that would have been a nice way to get RC's and regular releases in without having several sources.

Reply:
> You could try to fetchurl `https://github.com/php/php-src/archive/php-${version}.tar.gz` instead. It may yield a different result and you probably need to change the hash to get this different result.

That URL is what `fetchFromGitHub` uses internally.

Reply:
In `ttps://github.com/php/php-src/archive/php-${version}.tar.gz` missing file `pear/install-pear-nozlib.phar`

Reply:
Builded with this option - "--without-pear". Update?

Reply:
> > You could try to fetchurl `https://github.com/php/php-src/archive/php-${version}.tar.gz` instead. It may yield a different result and you probably need to change the hash to get this different result.
> 
> That URL is what `fetchFromGitHub` uses internally.

Oh, yeah. I just went to a release and copied the link and injected version variable there. But seems like they don't upload the tarballs there themselfs. Which makes me conclude that we should avoid them.

> Builded with this option - "--without-pear". Update?

I don't think we want to change that. I don't know how much pear is used nowadays? Anyone knows this? Because if it's not used, we can probably drop it. But not for this reason.

Reply:
Checking this method:
```
diff --git a/pkgs/development/interpreters/php/default.nix b/pkgs/development/interpreters/php/default.nix
index 5125ba0b0ea..cc310e22ab7 100644
--- a/pkgs/development/interpreters/php/default.nix
+++ b/pkgs/development/interpreters/php/default.nix
@@ -1,5 +1,5 @@
 # pcre functionality is tested in nixos/tests/php-pcre.nix
-{ config, lib, stdenv, fetchFromGitHub, autoconf, bison, libtool, pkgconfig, re2c
+{ config, lib, stdenv, fetchurl, fetchFromGitHub, autoconf, bison, libtool, pkgconfig, re2c
 , libmysqlclient, libxml2, readline, zlib, curl, postgresql, gettext
 , openssl, pcre, pcre2, sqlite
 , libxslt, libmcrypt, bzip2, icu, openldap, cyrus_sasl, libmhash, unixODBC
@@ -68,6 +68,10 @@ let
     let
       mysqlBuildInputs = optional (!mysqlndSupport) libmysqlclient;
       libmcrypt' = libmcrypt.override { disablePosixThreads = true; };
+      pear-nozlib = fetchurl {
+        url = "https://pear.php.net/install-pear-nozlib.phar";
+        sha256 = "19jizza0h7xxnnnw8wpsw6qr62yab9vzhd6qac7dlgwdw6vv5kka";
+      };
     in stdenv.mkDerivation {

       inherit version;
@@ -232,6 +236,10 @@ let
         ./buildconf --force
       '';

+      preInstall = ''
+        cp ${pear-nozlib} $TMPDIR/source/pear/install-pear-nozlib.phar
+      '';
+
       postInstall = ''
         test -d $out/etc || mkdir $out/etc
         cp php.ini-production $out/etc/php.ini

```

Reply:
Worked:
```
./result/bin/pear help version                                                                                                                                                                            
PEAR Version: 1.10.9
PHP Version: 7.2.24
Zend Engine Version: 3.2.0
Running on: Linux elven.pw 5.3.11-hardened #1-NixOS SMP Tue Nov 12 18:28:31 UTC 2019 x86_64
```
File `install-pear-nozlib.phar` install this package [PEAR](https://pear.php.net/package/PEAR/)